docs/sdks/c-sharp/index-v2.md

```diff
@@ -0,0 +1,860 @@
+| [`OnDelete` callback](#callback-ondelete)     | Register a callback to run whenever a row is deleted from the client cache.  |
+| [`OnUpdate` callback](#callback-onupdate)     | Register a callback to run whenever a subscribed row is replaced with a new version.  |
+
+#### Type `Row`
+
+```csharp
+class RemoteTableHandle<EventContext, Row>
+{
+    /* members */
+}
+```
+
+The type of rows in the table.
+
+#### Property `Count`
+
+```csharp
+class RemoteTableHandle
+{
+    public int Count;
+}
+```
+
+The number of rows of this table resident in the client cache, i.e. the total number which match any subscribed query.
+
+#### Method `Iter`
+
+```csharp
+class RemoteTableHandle
+{
+    public IEnumerable<Row> Iter();
+}
+```
+
+An iterator over all the subscribed rows in the client cache, i.e. those which match any subscribed query.
+
+#### Callback `OnInsert`
+
+```csharp
+class RemoteTableHandle
+{
+    public delegate void RowEventHandler(EventContext context, Row row);
+    public event RowEventHandler? OnInsert;
+}
+```
+
+The `OnInsert` callback runs whenever a new row is inserted into the client cache, either when applying a subscription or being notified of a transaction. The passed [`EventContext`](#type-eventcontext) contains an [`Event`](#record-event) which can identify the change which caused the insertion, and also allows the callback to interact with the connection, inspect the client cache and invoke reducers. Newly registered or canceled callbacks do not take effect until the following event.
+
+TODO: Add example of registering and unregistering callback
+
+#### Callback `OnDelete`
+
+```csharp
+class RemoteTableHandle
+{
+    public delegate void RowEventHandler(EventContext context, Row row);
+    public event RowEventHandler? OnDelete;
+}
+```
+
+The `OnDelete` callback runs whenever a previously-resident row is deleted from the client cache. Newly registered or canceled callbacks do not take effect until the following event.
+
+TODO: Add example of registering and unregistering callback
+
+#### Callback `OnUpdate`
+
+```csharp
+class RemoteTableHandle
+{
+    public delegate void RowEventHandler(EventContext context, Row row);
+    public event RowEventHandler? OnUpdate;
+}
+```
+
+The `OnUpdate` callback runs whenever an already-resident row in the client cache is updated, i.e. replaced with a new row that has the same primary key. Newly registered or canceled callbacks do not take effect until the following event.
+
+TODO: Add example of registering and unregistering callback
+
+### Unique constraint index access
+
+For each unique constraint on a table, its table handle has a property which is a unique index handle and whose name is the unique column name. This unique index handle has a method `.Find(Column: value)`. If a `Row` with `value` in the unique column is resident in the client cache, `.Find` returns it. Otherwise it returns null.
```

Comment:
This needs an example, also should say `.Find(Column value)`.

Module-side I just unique indexes as a class. 

https://github.com/clockworklabs/spacetime-docs/pull/193/files#diff-84b02ce60e346c81f89c934fbaa3096d08eac9ee7c3f2cac7372efa3127ef128R450

Module-side we kind of have to fake it --the base class doesn't actually have all the methods I pretend it does, due to nullable generic struct/class problems. I don't know if that's the case SDK-side.